ostelco-ios-client/ViewControllers/Home/ApplePayViewController.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  ApplePayViewController.swift
+//  ostelco-ios-client
+//
+//  Created by Prasanth Ullattil on 26/04/2019.
+//  Copyright © 2019 mac. All rights reserved.
+//
+
+import UIKit
+import Stripe
+import Siesta
+
+// Default implementaion for the PKPaymentAuthorizationViewControllerDelegate methods.
+class ApplePayViewController: UIViewController, ApplePayDelegate {
+
+    // Properties for ApplePayDelegate protocol
+    var shownApplePay = false
+    var authorizedApplePay = false
+    var purchasingProduct: Product?
+    var applePayError: ApplePayError?
+
+    func paymentAuthorizationViewController(_ controller: PKPaymentAuthorizationViewController,
+                                            didAuthorizePayment payment: PKPayment,
+                                            handler completion: @escaping (PKPaymentAuthorizationResult) -> Void) {
+        authorizedApplePay = true
+        let product = purchasingProduct!
+        // Create Stripe Source.
+        STPAPIClient.shared().createSource(with: payment) { (source: STPSource?, error: Error?) in
+            guard let source = source, error == nil else {
+                debugPrint(error!)
+                completion(PKPaymentAuthorizationResult(status: .failure, errors: [error!]))
+                self.showAlert(title: "Failed to create stripe source", msg: "\(error!.localizedDescription)")
+                return
+            }
+            // Call Prime API to buy the product.
+            APIManager.sharedInstance.products.child(product.sku).child("purchase").withParam("sourceId", source.stripeID).request(.post)
+                .onProgress({ progress in
+                    debugPrint("Progress %{public}@", "\(progress)")
```

Comment:
You can do this just as `"Progress: \(progress)"` with `debugPrint` since it's not being logged when not actively debugging the way `os_log` can be

Reply:
done